### DISCUSSION TITLE
Can I install frameworks manually?

### DISCUSSION THREAD

**Discussion by qminds** [iBotPeaches/Apktool#2667] [Q&A] [answered]
What if I just manually copy a framework file to the frameworks directory instead of running Apktool with the `--install-framework` argument? Will it work the same way, or the `--install-framework` is mandatory (i.e. this parameter does more than just copying and renaming the framework file)?

**Comment by iBotPeaches** [comment, marked as answer]
Apologies for missing this. You can manually copy them, the only thing it does is name them in the preferred way. Which is `{pkgId}-{?tagId}.apk`. pkgId may be difficult to do manually as you read the `resources.arsc` table to find the ResPackage with most ResSpecs.
